### DISCUSSION TITLE
Help remove an ad

### DISCUSSION THREAD

**Discussion by 54m4d** [Daksh777/SpotifyNoPremium#22] [Q&A] [answered]
Hello Daksh

Could you help me find the selector for this homepage ad and tell me how to eliminate it?

<details>
<summary>Here is the ad:</summary>
<img src=https://user-images.githubusercontent.com/52612348/132091072-21b90333-8a6a-477f-8bd4-310fe684ba90.png>
</details>

So how can i start removing the ad? i do not use Spicetify at this moment due to some issues, sorry about that

**Comment by Daksh777** [comment, marked as answer]
This ad should not appear if you're using BlockTheSpot. Since I do not see this ad, I can only help by seeing the screenshot. 
So, you may have to try with the class `k6onUQZSeZIsonJ4cSpK` and set it to `display: none`

**Reply by 54m4d** [reply]
I don't don't use BTS since i don't have audio ads, i searched for `k6onUQZSeZIsonJ4cSpK` in the css file, but i could not find it, the only thing i found was `euOnte9wvOF0D_SGxEZ9` and i have added `display: none` to it but it did not remove the ad, does that mean it is the wrong selector?

**Reply by Daksh777** [reply]
Unfortunately, I do not see any such ad on the home page even after removing BTS. 
You will have to do some hit and trial in order to get the correct selector. You should try targeting the parent element and if that doesn't work, go to the child elements.

![image](https://user-images.githubusercontent.com/43648146/132449731-59295cdb-df83-41af-819f-4954fc87e7e1.png)
For example, this ad placeholder which was the parent element and I removed it by using
```
.snMc3rl8H2pM9cqkTpQ6 {
  display: none !important
}
```

**Reply by 54m4d** [reply]
I searched for the selectors that are above the one that is selected in the screenshot, but i only found this one `euOnte9wvOF0D_SGxEZ9`

I added the argument but it did not remove ad

<details>
<summary>Here is how it looks:</summary>
<img src=https://user-images.githubusercontent.com/52612348/132376504-b63bebe7-c0cf-416e-aa88-26c8e9a3ef97.png>
</details>

Does it look correct?

**Reply by Daksh777** [reply]
Try adding `!important` and see if it makes a difference.
Also, are you sure the CSS you're editing is actually taking effect? Try editing the CSS in live mode and see which selector works and then apply it using Spicetify
![image](https://user-images.githubusercontent.com/43648146/132625790-b9241016-dc15-4626-98e3-876f34cfa8fd.png)

**Reply by 54m4d** [reply]
Maybe i forgot to mention that i don't use Spicetify or BTS at the moment, i only tried to edit the selector i mentioned above in the xpui.css file.

<details>
<summary>Here is the unmodified part of the css where the selector i mentioned above is highlighted 3 times in the xpui.css file:</summary>
<img src=https://user-images.githubusercontent.com/52612348/132629991-b362ba7f-e2b6-4cf6-be20-38974a79895c.png>
</details>

Now could you write me what i need to add and where i need to add it based on this screenshot? first i added `display:none` and it did not work but i'm not sure if it suppose to be `display:none` or `display: none`

I will also try editing in live mode as soon as the ad appears again because the ad comes and goes, so at this moment of writing i don't have it but it will come back soon.

**Reply by Daksh777** [reply]
Unfortunately, it's too difficult for me to actually tell which class belongs to the particular ad you're talking about.
The only thing you can do is use inspect element and edit the classes live and see which class belongs to the ad.
Also, writing `display:none` or `display: none` does not make any difference so it should be okay.

**Reply by 54m4d** [reply]
Thanks!

I will try your tip once the ad shows up again.

**Reply by 54m4d** [reply]
I can finally confirm the correct selector by trying a live demo

<details>
<summary>Live demo</summary>
<img src=https://user-images.githubusercontent.com/52612348/132971699-546fd5b5-cead-42ea-b8b0-31e701fa378b.GIF>
</details>

But the issue is that i can't find this selector in the css file and my goal is not to use Spicetify and BTS 🤔

**Reply by Daksh777** [reply]
It's unlikely that the selector is not in the file. Are you sure you're checking in the same file as mentioned in the dev tools?
![image](https://user-images.githubusercontent.com/43648146/132982650-71fc14bd-2f6b-4ea5-af0c-ec669ca73477.png)

**Reply by 54m4d** [reply]
Oh, i forget to include that part in the screenshot

<details>
<summary>Does this look correct to you?</summary>
<img src=https://user-images.githubusercontent.com/52612348/132994410-63712ded-4082-4209-bd76-a0de7653331b.png>
</details>

**Reply by Daksh777** [reply]
That's right, the file is in `C:\Users\USERNAME\AppData\Roaming\Spotify\Apps\xpui`, specifically for the homepage ad.
You may try adding `display: none` like this: From `.k6onUQZSeZIsonJ4cSpK{max-height:350px;min-height:270px}` to `.k6onUQZSeZIsonJ4cSpK{max-height:350px;min-height:270px;display:none}`

**Reply by 54m4d** [reply]
Good, i will try this as soon as the ad appears again, because it comes and goes.

**Reply by 54m4d** [reply]
Now the ad came back, and you are right!, i just had to add `!important` to your last example as you mentioned in the past, and after restarting i saw no home ads, and i hope it stays like this.

Thank you very much for your patience with me and helping me remove this element 👍

**Comment by 54m4d** [comment]
@Daksh777 i usually don't use the option called `overwrite_assets` could you tell me what it does? and what's the differance between enabling it and disabling it? i couldn't find any info about it in Spicetify. (by the name i can tell it replaces the assets, but i don't know what it does exactly)

**Reply by Daksh777** [reply]
Please ask this question in a separate Spicetify discussion next time.
I don't exactly know what does disabling and enabling `overwrite_assets` does but from what it seems like in the [source code](https://github.com/khanhas/spicetify-cli/blob/5e4e99b5327a746e190c7d4f37378afa17a92989/src/cmd/watch.go#L47-L60), it seems to be running the [`upateAssets`](https://github.com/khanhas/spicetify-cli/blob/5e4e99b5327a746e190c7d4f37378afa17a92989/src/cmd/apply.go#L138-L140) function with an error check if I'm not wrong
![image](https://user-images.githubusercontent.com/43648146/134465553-e0fd879e-aa9a-4864-9c46-7ea9674a0ab0.png)
![image](https://user-images.githubusercontent.com/43648146/134465611-caec819d-3a4d-4fd8-90ad-ece302d71e17.png)

It'd be better if you ask this question in the main repo itself as I don't have a solid answer.
